polaris-react/src/styles/shared/_breakpoints.scss

```diff
@@ -1,41 +1,55 @@
-// Reduce breakpoints
+/* stylelint-disable unit-disallowed-list */
 
-@function breakpoint($value) {
+
+@function breakpoint($value, $inclusive: true) {
   $value-unit: unit($value);
+  // Down media condition breakpoints are being subtracted by 0.05px to prevent
+  // them from overwriting up media queries. We experimented with multiple offsets
+  // and felt that 0.05px would be the safest across different pixel densities.
+  $offset: -0.003125em; // -0.05px
 
   @if $value-unit == 'em' {
-    @return $value;
+    @return $value + if($inclusive, 0, $offset);
   } @else if $value-unit == 'px' {
-    @return $value / 16px * 1em;
+    @return ($value / 16px * 1em) + if($inclusive, 0, $offset);
   } @else {
     @error 'The $value passed into breakpoint() must be a pixel or em value. Got "#{$value}"';
   }
 }
 
-$page-max-width: $layout-width-primary-max + $layout-width-secondary-max +
-  $layout-width-inner-spacing-base;
+@function breakpoints-up($value) {
+  @return '(min-width: #{breakpoint($value)})';
+}
+
+@function breakpoints-down($value, $inclusive: false) {
+  @return '(max-width: #{breakpoint($value, $inclusive)})';
+}
```

Comment:
While reviewing these `mixins to variables` updates we caught multiple typos and find/replace issues (human error) that justified temporarily introducing `breakpoints-up` and `breakpoints-down` Scss functions. The breakpoint functions follow the same mobile first strategy used in the `gen-media-queries` script where `down` conditions apply a `-0.05px` offset to avoid conflicts with `up` conditions. One benefit of this design pattern is we can now clearly see `down` conditions that are not following our recommended approach as they will be explicitly defined with `$inclusive: false`

e.g. `$some-legacy-breakpoint: breakpoints-down(555px, $inclusive: false)`